src/components/year_month.rs

```diff
@@ -2,14 +2,24 @@
 
 use std::str::FromStr;
 
+use tinystr::TinyAsciiStr;
+
 use crate::{
     components::calendar::Calendar,
+    fields::FieldMap,
     iso::{IsoDate, IsoDateSlots},
     options::ArithmeticOverflow,
-    TemporalError, TemporalResult, TemporalUnwrap,
+    utils::{self, pad_iso_year},
+    TemporalError, TemporalFields, TemporalResult, TemporalUnwrap,
+};
+
+use super::{
+    calendar::{CalendarDateLike, GetTemporalCalendar},
+    Duration,
 };
 
-use super::calendar::GetTemporalCalendar;
+// Subset of `TemporalFields` representing just the  `YearMonthFields`
+pub struct YearMonthFields(i32, u8);
```

Comment:
question: is there a primary use case for this?

I think it's mostly used to implement `impl From<YearMonthFields> for TemporalFields`, and I'm sort of curious if it would just be better to implement `From<YearMonth>`

Reply:
Yes I think so, I can take a look to see if we can just implement From

Reply:
This is used for add_or_subtract_duration to get the fields of YearMonth then generate a Date object